ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml

```diff
@@ -77,59 +89,104 @@ StackView {
         pop(initialItem, StackView.Immediate)
     }
 
+    // This method will be called from the outsite from a different section like Airdrop or Permissions
     function openNewTokenForm(isAssetView) {
         resetNavigation()
 
-        if(d.isTokenOwnerDeployed) {
+        if(root.isAdminOnly) {
+            // Admins can only see the initial tokens page. They cannot mint
+            root.push(mintedTokensViewComponent, StackView.Immediate)
+            return
+        }
+
+        if(root.arePrivilegedTokensInProgress || root.arePrivilegedTokensFailed) {
+            // If Owner and TMaster tokens deployment action has been started at least ones, but still without success
+            root.push(mintedTokensViewComponent, StackView.Immediate)
+            return
+        }
```

Comment:
There is the same action it those cases, also the default one is the same. Cannot it be simplified a bit?

Reply:
Yes, it could be simplified but I did it expressly like this to be more descriptive. Since there are lots of different use cases I think it's easy to read it separately that having less code in this case. WDYT?